database/migrations/default/1750435376415_ems_person_match_auto/up.sql

```diff
@@ -459,7 +465,36 @@ BEGIN
 END;
 $function$;
 
--- delete this trigger which is now redundant
+
+--
+-- This is a new function that triggers EMS re-matching when a person record is inserted or updated
+--
+CREATE
+OR REPLACE FUNCTION public.people_dispatch_ems_match() RETURNS trigger LANGUAGE plpgsql AS $function$
+DECLARE
+    person_crash_pk integer;
+BEGIN
+    raise debug '**function: people_dispatch_ems_match **';
+    
+    SELECT units.crash_pk INTO person_crash_pk FROM units where units.id = NEW.unit_id;
+    
+    raise debug 'Dispatching update event to EMS records matched to crash_pk %', person_crash_pk;
+
+    UPDATE ems__incidents 
+    SET 
+        _match_event_name = 'handle_person_record_created_or_updated'
+    WHERE
+       ems__incidents.crash_pk = person_crash_pk;
+    RETURN null;
+END;
+$function$;
+
+DROP TRIGGER IF EXISTS people_dispatch_ems_match_trigger ON people;
+CREATE TRIGGER people_dispatch_ems_match_trigger AFTER
+INSERT OR UPDATE ON people FOR EACH ROW
```

Comment:
Something obvious that I completely missed in the parent PR: if we try to match an EMS record to a person record when a crash is inserted, it will never match because people records are created by the ETL after the crash is created. 

So prior to this change, an EMS-person level match was only ever possible if a crash's time or location were edited. A match would never happen when the crash was inserted 🙃

Reply:
Nice catch! I'm sad that I didn't catch this either 😅 Lots of moving parts!